### PR TITLE
Remove use of azure next link

### PR DESCRIPTION
**Pull request by AlitzelMendez**
Fixing breaking change caused by: https://github.com/Azure/typespec-azure/pull/1886
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

**Review by weidongxu-microsoft — COMMENTED** [review]
@tadelesh Does TCGC at present handle the 2 `nextLink` annotation the same way?
If yes, I guess we can remove it here.

@AlitzelMendez I assume this PR depends on the merge on your typespec-azure-core PR? https://github.com/Azure/typespec-azure/pull/1886

**Comment by weidongxu-microsoft** [comment]
Got it. So this PR would be read-to-merge after the PR to typespec-azure get merged.

**Comment by AlitzelMendez** [comment]
> @tadelesh Does TCGC at present handle the 2 `nextLink` annotation the same way? If yes, I guess we can remove it here.
> 
> @AlitzelMendez I assume this PR depends on the merge on your typespec-azure-core PR? [Azure/typespec-azure#1886](https://github.com/Azure/typespec-azure/pull/1886)

I would prefer if we can merge this first so I can have a green build in my other pull request, as we are running the E2E it fails as we are removing something that is still use in here

@markcowl could you please take a look? 😄

**Review by markcowl — APPROVED** [review]

**Comment by markcowl** [comment]
@tadelesh @weidongxu-microsoft  Because we verify cadl-ranch as part of ci checks in typespec-azure, this PR has to be merged before removing the decorator there.  Can one of you or the other code owners approve this?

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
approved and merged

**Pull request merged**
